### PR TITLE
A few buffer logic fixes

### PR DESCRIPTION
**Pull request by lasalvavida**
These fixes come from integrating pipeline into 3D Tiles processing and having some conversion problems.

Removes an unnecessary 'if' wrapping when copying data chunks from removeUnusedVertices. It shouldn't ever be copying data past the end of the segment, so if it is we want it to fail because something else is wrong.

uninterleaveAndPackBuffers had an issue with naming collision in bufferViews, and this should fix that.

@lilleyse Could you take a look at this when you get a chance and maybe run a few random sample models through it to make sure they don't break?

### CODE REVIEW

**Comment by lasalvavida** [comment]
I have some test failures that need fixing first.

**Comment by lasalvavida** [comment]
That should fix the tests, this is ready for a look.

File: lib/uninterleaveAndPackBuffers.js [hunk 1]
```diff
@@ -98,7 +121,9 @@ function packAccessors(gltf, accessorsToPack, sourceBuffer, packBuffer, bufferVi
         var offset = byteOffset;
         var count = accessor.count;
         accessor.byteStride = 0;
-        accessor.bufferView = bufferViewId;
+        // Store as temp for now, finalize later
+        var extras = getPipelineExtras(accessor);
```
Comment (lilleyse) [inline lib/uninterleaveAndPackBuffers.js]: Is this comment something you will address before this PR is done?

Comment (lasalvavida) [inline lib/uninterleaveAndPackBuffers.js]: This refers to the fact that it temporarily stores the new bufferViewId in extras and then permanently assigns it at the end to avoid naming conflicts. It isn't a TODO.

**Comment by lasalvavida** [comment]
@lilleyse This is ready for a look. It refactors removeUnusedIndices to cover all of the possible edge cases and adds a test for the cross-dependency issue I was having. 

The tilesets I was testing on process correctly now, and the BrainStem model still has the same result from merging the duplicates; all the old tests still pass.

I added uint32 index support for the sponza model as well.

**Comment by JoshuaStorm** [comment]
@lasalvavida Build is currently failing

**Comment by lasalvavida** [comment]
@JoshuaStorm Yup, sorry about that, should be fixed now

File: lib/byteLengthForComponentType.js [hunk 2]
```diff
@@ -25,6 +28,7 @@ function byteLengthForComponentType(componentType) {
         case WebGLConstants.UNSIGNED_SHORT:
             return 2;
         case WebGLConstants.FLOAT:
+        case WebGLConstants.UNSIGNED_INT:
```
Comment (lilleyse) [inline lib/byteLengthForComponentType.js]: Also update `readBufferComponentType` and `writeBufferComponentType` with the new type.

Comment (lasalvavida) [inline lib/byteLengthForComponentType.js]: Updated

File: lib/readBufferComponentType.js [hunk 3]
```diff
@@ -14,6 +14,8 @@ function readBufferComponentType(buffer, componentType, byteOffset) {
             return buffer.readInt16LE(byteOffset);
         case WebGLConstants.UNSIGNED_SHORT:
             return buffer.readUInt16LE(byteOffset);
+        case WebGLConstants.UNSIGNED_INT:
+            return buffer.readUInt32LE(value, byteOffset);
```
Comment (lilleyse) [inline lib/readBufferComponentType.js]: Remove `value'.

**Comment by lilleyse** [comment]
I was about to comment that the sponza model wasn't loading in Cesium, but the recent change fixes that, so that's good!

File: lib/readBufferComponentType.js (hunk 3 above)
Comment (lasalvavida) [inline lib/readBufferComponentType.js]: Whoops, thanks for catching that

File: lib/createAccessorUsageTables.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                        for (attribute in attributes) {
+                            if (attributes.hasOwnProperty(attribute)) {
+                                attributeAccessorId = attributes[attribute];
+                                // Make sure the attribute accessors haven't been assigned somewhere else
+                                var attributeTableIndex = accessorTableMapping[attributeAccessorId];
+                                if (defined(attributeTableIndex) && attributeTableIndex !== tableIndex) {
+                                    // Merge them into this one if they have been
+                                    mergeUsageTables(tables, accessorTableMapping, tableIndex, attributeTableIndex);
+                                }
+                                table.attributeAccessors[attributeAccessorId] = true;
```
Comment (JoshuaStorm) [inline lib/createAccessorUsageTables.js]: Is there any reason you wouldn't be able to use the `attributeAccessors` attribute type as the value instead of just setting it to true? In order to use `createAccessorUsageTables` in PR #100 I would need the attribute types associated with the accessors and it seems like that would be a good spot to sneak them in.
Thanks!

**Comment by lilleyse** [comment]
Tested a few models, seems good to me.

File: lib/createAccessorUsageTables.js (hunk 4 above)
Comment (lasalvavida) [inline lib/createAccessorUsageTables.js]: the attributeAccessor value is now assigned to be the semantic

Comment (JoshuaStorm) [inline lib/createAccessorUsageTables.js]: Thanks! My bad on the mixed up terminology

**Comment by lasalvavida** [comment]
@lilleyse This should be ready, merge whenever you're ready
